### PR TITLE
GlueSQL.js - support both web browsers and Node.js

### LINKED ISSUE #489: Migrate the `gluesql-js` repository into the workspace
ref: <https://github.com/gluesql/gluesql/issues/480>

We hope to add workspaces like `[gluesql(Rust), gluesql-js, gluesql-py]` in the future.

### PR DESCRIPTION
**Pull request by panarch**
Add gluesql-js/ workspace.
gluesql-js/README.md - npm package intro document
gluesql-js/web/README.md - wasm build guide document
Current GlueSQL.js supports only MemoryStorage backend.
Basic example usage codes can be found in gluesql-js/examples/
Three binary files are provided.
* gluesql.js - for webpack & js modules
* gluesql.rollup.js - for rollup
* gluesql.node.js - for node.js

resolve #489

### CODE REVIEW

**Comment by ever0de** [comment]
link #489

**Comment by ever0de** [comment]
Let's proceed with the review.
The goal is to do it at least before 5/1.

**Review by ever0de — APPROVED** [review]
LGTM :)

File: gluesql-js/web/src/lib.rs [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+        let cell = Rc::clone(&self.storage);
+
+        future_to_promise(async move {
+            let queries = parse(&sql).map_err(|error| JsValue::from_str(&format!("{error}")))?;
+
+            let mut payloads = vec![];
+            let mut storage: MemoryStorage = cell.replace(None).unwrap();
+
+            for query in queries.iter() {
+                let statement = translate(query); //.map_err(|error| JsValue::from_str(&format!("{error}")))?;
```
[the remark this answers is not in the document]
Comment (panarch) [reply gluesql-js/web/src/lib.rs]: Thanks 👍

**Pull request merged**